tests/specs/infinite-gas.k

```diff
@@ -0,0 +1,55 @@
+requires "evm.md"
+
+module INFINITE-GAS
+    imports EVM
+
+
+ // Symbolic Gas
+
+    syntax Int ::= #gas ( Int , Int , Int )  [function]  // startGas, nonMemory, memory
+ //-----------------------------------------------------------------------------------
+
+ //
+ // Infinite Gas
+ // Here we use the construct `#gas` to count _up_ gas used instead of counting down.
+ // This allows (i) computing final gas used, and (ii) never stopping because of out-of-gas.
+ //
+
+    syntax Int ::= #gas ( Int ) [function, functional, no-evaluators]
+ // -----------------------------------------------------------------
+    rule #gas(_, _, _) => #gas(0)
+
+    rule #gas(G) -Int G' => #gas(G +Int G') [simplification]
+
+    rule #gas(_)  <Int I => false requires I <=Int pow256 [simplification]
+    rule #gas(_) <=Int I => false requires I <=Int pow256 [simplification]
+    rule #gas(_) >=Int I => true  requires I <=Int pow256 [simplification]
+    rule I <=Int #gas(_) => true  requires I <=Int pow256 [simplification]
+
+    rule #gas(_)  <Int _I => false [concrete(_I), simplification]
```

Comment:
Isn't this block of rules redundant to the rules above? And contradictory?

Reply:
I'm not sure how it's redundant or contradictory. Here we're saying that `#gas(_)` is not smaller than any particular number that is nameable as a concrete number. It's just a heuristic for saying that `#gas(_)` is larger than any particular number, this heuristic is for any number that is concrete, which means it's some particular integer.